modrinth/install.go

```diff
@@ -0,0 +1,224 @@
+package modrinth
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/comp500/packwiz/core"
+	"github.com/spf13/cobra"
+)
+
+var modSiteRegex = regexp.MustCompile("modrinth\\.com/mod/([^/]+)/?$")
+var versionSiteRegex = regexp.MustCompile("modrinth\\.com/mod/([^/]+)/version/([^/]+)/?$")
+
+// installCmd represents the install command
+var installCmd = &cobra.Command{
+	Use:     "install [mod]",
+	Short:   "Install a mod from a modrinth URL, slug, ID or search",
+	Aliases: []string{"add", "get"},
+	Args:    cobra.ArbitraryArgs,
+	Run: func(cmd *cobra.Command, args []string) {
+		pack, err := core.LoadPack()
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+
+		if len(args) == 0 || len(args[0]) == 0 {
+			fmt.Println("You must specify a mod.")
+			os.Exit(1)
+		}
+
+		// If there are more than 1 argument, go straight to searching - URLs/Slugs should not have spaces!
+		if len(args) > 1 {
+			err = installViaSearch(strings.Join(args, " "), pack)
+			if err != nil {
+				fmt.Printf("Failed installing mod: %s\n", err)
+				os.Exit(1)
+			}
+			return
+		}
+
+		//Try interpreting the arg as a version url
+		matches := versionSiteRegex.FindStringSubmatch(args[0])
+		if matches != nil && len(matches) == 3 {
+			err = installVersionById(matches[2], pack)
+			if err != nil {
+				fmt.Printf("Failed installing mod: %s\n", err)
+				os.Exit(1)
+			}
+			return
+		}
+
+		//Try interpreting the arg as a modId or slug.
+		//Modrinth transparently handles slugs/mod ids in their api; we don't have to detect which one it is.
+		var modStr string
+
+		//Try to see if it's a site, if extract the id/slug from the url.
+		//Otherwise, interpret the arg as a id/slug straight up
+		matches = modSiteRegex.FindStringSubmatch(args[0])
+		if matches != nil && len(matches) == 2 {
+			modStr = matches[1]
+		} else {
+			modStr = args[0]
+		}
+
+		mod, err := fetchMod(modStr)
+
+		if err == nil {
+			//We found a mod with that id/slug
+			err = installMod(mod, pack)
+			if err != nil {
+				fmt.Printf("Failed installing mod: %s\n", err)
+				os.Exit(1)
+			}
+			return
+		} else {
+			//This wasn't a valid modid/slug, try to search for it instead:
+			//Don't bother to search if it looks like a url though
+			if !strings.Contains(args[0], "modrinth.com") {
+				err = installViaSearch(args[0], pack)
+				if err != nil {
+					fmt.Printf("Failed installing mod: %s\n", err)
+					os.Exit(1)
+				}
+			}
+		}
+	},
+}
+
+func installViaSearch(query string, pack core.Pack) error {
+	mcVersion, err := pack.GetMCVersion()
+	if err != nil {
+		return err
+	}
+
+	searchResult, err := getFirstModIdViaSearch(query, mcVersion)
```

Comment:
It'd be great if you could reuse the mod selection code I have in `searchCurseforgeInternal` for this - often a partial match won't get you the right mod!
As an example searching `Oh no` gets you `Oh The Biomes You'll Go: 'An Adventure & Exploration mod like no other with unique, magical, realistic, and detailed biomes!'.`

Reply:
That's probably a good idea!